op-service/ioutil/gzip.go

```diff
@@ -9,26 +9,57 @@ import (
 	"strings"
 )
 
+// gzipReadCloser is a struct that closes both the gzip.Reader and the underlying io.Closer.
+type gzipReadCloser struct {
```

Comment:
These seem like generically useful utilities rather than being specifically tied to gzip.  I'd be tempted to move them to their own file and rename to `WrappedReadCloser` or something like that (definitely open to ideas for better names).

And to make it usable from other packages it would be good to expose constructor functions like:

```
func NewWrappedReadCloser(r io.ReadCloser, c io.Closer) *WrappedReadCloser
```

Reply:
In the current PR, I will focus on rename and adding the constructor function. However, I'll move the struct and the new constructor to a separate file in a follow-up PR. This approach should keep the scope of this PR clear and make the code review process easier.

Please let me know if that works for you!

Reply:
Given we're adding these new types in this PR I think it's worth moving them to a new file straight away. Otherwise it looks great.

Reply:
Fine! I would like to make file name compress_closer.go on the same directory
Do you have better idea for the name?

Reply:
I'd probably call it `wrapped_closer.go` and yep in the same directory makes sense.

Reply:
@ajsutton I made it !
could you check this?